python/ray/serve/config.py

```diff
@@ -219,6 +227,14 @@ class AutoscalingConfig(BaseModel):
         default=30.0, description="How long to wait before scaling up replicas."
     )
 
+    # New fields for custom autoscaling
+    agg_function: str = "mean"  # "mean", "min", "max", "sum" how to aggregate metrics within look_back_period_s
+    custom_policy: Optional[
+        AutoscalingPolicyConfig
+    ] = None  # this policy is deployment scoped
+    # List of []
+    prometheus_custom_metrics: Optional[List[Tuple[str, Optional[str]]]] = None
```

Comment:
The first tuple field is the metric_name, the second is the optional PromQL expression. If no PromQL is specified, defaults to fetching the last metric value.